api/insight/apimiddleware.go

```diff
@@ -91,3 +109,161 @@ func (c *insightApiContext) PostBroadcastTxCtx(next http.Handler) http.Handler {
+	to, ok := r.Context().Value(ctxTo).(int)
+	if !ok {
+		apiLog.Errorf("to param not set")
+		return int64(0), false
+	}
+	return int64(to), true
+}
+
+// GetFromCtx retrieves the ctxFrom data ("from") from the request context.
+// If not set, the return value ok is false.
+func (c *insightApiContext) GetFromCtx(r *http.Request) (int64, bool) {
+	from, ok := r.Context().Value(ctxFrom).(int)
+	if !ok {
+		apiLog.Errorf("from param not set")
+		return int64(0), false
+	}
+	return int64(from), true
+}
+
+// GetNoAsmCtx retrieves the ctxNoAsm data ("noAsm") from the request context.
+// If not set, the return value is false.
+func (c *insightApiContext) GetNoAsmCtx(r *http.Request) bool {
+	noAsm, ok := r.Context().Value(ctxNoAsm).(bool)
+	if !ok {
+		return false
+	}
+	return noAsm
+}
+
+// GetNoScriptSigCtx retrieves the ctxNoScriptSig data ("noScriptSig") from the
+// request context. If not set, the return value is false.
+func (c *insightApiContext) GetNoScriptSigCtx(r *http.Request) bool {
+	noScriptSig, ok := r.Context().Value(ctxNoScriptSig).(bool)
+	if !ok {
+		return false
+	}
+	return noScriptSig
+}
+
+// GetNoSpentCtx retrieves the ctxNoSpent data ("noSpent") from the
+// request context. If not set, the return value is false.
+func (c *insightApiContext) GetNoSpentCtx(r *http.Request) bool {
+	noNoSpent, ok := r.Context().Value(ctxNoSpent).(bool)
+	if !ok {
+		return false
+	}
+	return noNoSpent
+}
+
+// PaginationCtx will parse the query parameters for from/to values.
+func (c *insightApiContext) PaginationCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		to, from := r.FormValue("to"), r.FormValue("from")
+		fromint, err := strconv.Atoi(from)
+		if err == nil {
+			ctx = context.WithValue(r.Context(), ctxFrom, fromint)
+		}
+		toint, err := strconv.Atoi(to)
+		if err == nil {
+			ctx = context.WithValue(ctx, ctxTo, toint)
+		}
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// Process params given in post body for an addrs endpoint
+func (c *insightApiContext) PostAddrsTxsCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req apitypes.InsightMultiAddrsTx
+		var from, to, noAsm, noScriptSig, noSpent int64
+		body, err := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if err != nil {
+			apiLog.Errorf("error reading JSON message: %v", err)
+			writeInsightError(w, fmt.Sprintf("error reading JSON message: %v", err))
+			return
+		}
+		err = json.Unmarshal(body, &req)
+		if err != nil {
+			apiLog.Errorf("Failed to parse request: %v", err)
+			writeInsightError(w, fmt.Sprintf("Failed to parse request: %v", err))
+			return
+		}
+		// Successful extraction of Body JSON
+		ctx := context.WithValue(r.Context(), ctxAddress, req.Addresses)
+
+		if req.From != "" {
+			from, err = req.From.Int64()
+			if err != nil {
+				apiLog.Errorf("Error converting 'from' to Int.  Expecting Integer.")
```

Comment:
Since these depend on input from the client, it's trivial to fill up the logs with this.  Please make this Debugf, or perhaps check the error itself to only make an Errorf/Warnf in certain circumstances that indicate a server error rather than a bad request.

Reply:
I removed all logging references that were related to user data.  Agreed, not useful unless something is happening that should never happen.